src/OrchardCore.Modules/OrchardCore.Notifications/Migrations/NotificationMigrations.cs

```diff
@@ -31,4 +31,18 @@ await SchemaBuilder.AlterIndexTableAsync<NotificationIndex>(table => table
 
         return 1;
     }
+
+    public async Task<int> UpdateFrom1Async()
+    {
+        await SchemaBuilder.AlterIndexTableAsync<NotificationIndex>(table => table
+            .CreateIndex("IDX_NotificationIndex_UserId",
```

Comment:
Is the index used somewhere else?

Reply:
This add a SQL index not SQL Table. This index will be used by multiple queries to improve SQL performance.

Reply:
@hishamco unless you see an issue with the PR, please approve.

Reply:
Sure, I just added a comment waiting for a use case for your index, I'm not sure if it's good to add something without using it in the framework

Reply:
I did explain https://github.com/OrchardCMS/OrchardCore/pull/16590#discussion_r1723782067 and you replaied with 👍 . This new index will be used by the database server to optimize queries that are generated by the framework.

Reply:
I understand what you mean bro :) what I'd like to point out is it feasible to add functionality without using it in the framework itself. I remembered @Piedone in another PR asked you about usage for one of the extensions that you added recently because it's not used somewhere in the framework

So I'm not against your PR :)

Reply:
I don't think you understanding what I am saying here. I am saying that this index is used by OC framework.

Reply:
>  I am saying that this index is used by OC framework

Where will be used?

Reply:
It's used by the database engine when this call is made

https://github.com/OrchardCMS/OrchardCore/blob/51f24d72211f8642976547e34aef018e05b941fa/src/OrchardCore.Modules/OrchardCore.Notifications/Drivers/NotificationNavbarDisplayDriver.cs#L39

Also here 

https://github.com/OrchardCMS/OrchardCore/blob/51f24d72211f8642976547e34aef018e05b941fa/src/OrchardCore.Modules/OrchardCore.Notifications/Controllers/AdminController.cs#L108

Reply:
There is already an index that includes the same columns. Why isn't that sufficient? 
https://github.com/OrchardCMS/OrchardCore/blob/b65ecce68c2519de1b6a89d4f23cdd3a22a5b165/src/OrchardCore.Modules/OrchardCore.Notifications/Migrations/NotificationMigrations.cs#L22-L28

Reply:
@gvkries because that index used starts with DocumentId then followed by NotificationId which isn't used by either query. So the database engines will not use that index when for these two queries.

Reply:
Interesting, thanks for the explanation.

Reply:
FYI: When SQL engines select an index, they prioritize included columns from left to right. The new index is likely to be used when querying by `DocumentId` and `UserId`, or by `DocumentId`, `UserId`, and `IsRead`, or by `DocumentId`, `UserId`, and `CreatedAtUtc`. Columns on the right can be optional, but including them increases the likelihood that the index will be selected for a query.

Reply:
Just a follow-up question, what about the `Content` column. Isn't that probably too large for an index anyway?

Reply:
It is large, but just under the limits. Check out the comments for the column length

https://github.com/OrchardCMS/OrchardCore/blob/51f24d72211f8642976547e34aef018e05b941fa/src/OrchardCore/OrchardCore.Notifications.Core/NotificationConstants.cs#L7-L13